src/app/ReadClient.cpp

```diff
@@ -799,9 +799,8 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
                 attributePath.mListOp = ConcreteDataAttributePath::ListOperation::ReplaceAll;
             }
 
-            if (attributePath ==
-                ConcreteDataAttributePath(kRootEndpointId, Clusters::IcdManagement::Id,
-                                          Clusters::IcdManagement::Attributes::OperatingMode::Id))
+            if (attributePath.MatchesConcreteAttributePath(ConcreteDataAttributePath(
```

Comment:
is there any way to devise a unit test that would have failed on the previous code but would succeed now?

Reply:
Added unit tests to cover the code in this CL. Will have more in https://github.com/project-chip/connectedhomeip/pull/31809.